sdk/attestation/azure-security-attestation/src/main/java/com/azure/security/attestation/AttestationAdministrationAsyncClient.java

```diff
@@ -0,0 +1,473 @@
+ * to verify that they are authorized to perform the operation specified.
+ *</p>
+ *
+ */
+@ServiceClient(builder = AttestationClientBuilder.class, isAsync = true)
+public final class AttestationAdministrationAsyncClient {
+    private final SigningCertificatesImpl signingCertificatesImpl;
+    private final PoliciesImpl policyImpl;
+    private final PolicyCertificatesImpl certificatesImpl;
+    private final ClientLogger logger;
+    private final AttestationTokenValidationOptions tokenValidationOptions;
+    private final AtomicReference<List<AttestationSigner>> cachedSigners;
+    private static final SerializerAdapter SERIALIZER_ADAPTER = JacksonAdapter.createDefaultSerializerAdapter();
+
+    /**
+     * Initializes an instance of Attestations client.
+     *
+     * @param clientImpl the service client implementation.
+     */
+    AttestationAdministrationAsyncClient(AttestationClientImpl clientImpl, AttestationTokenValidationOptions tokenValidationOptions) {
+        this.signingCertificatesImpl = clientImpl.getSigningCertificates();
+        this.policyImpl = clientImpl.getPolicies();
+        this.certificatesImpl = clientImpl.getPolicyCertificates();
+        this.tokenValidationOptions = tokenValidationOptions;
+        this.logger = new ClientLogger(AttestationAdministrationAsyncClient.class);
+        this.cachedSigners = new AtomicReference<>(null);
+    }
+//region Get Attestation Policy.
+    /**
+     * Retrieves the current policy for an attestation type.
+     *
+     * @param attestationType Specifies the trusted execution environment whose policy should be retrieved.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return the attestation policy expressed as a string.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<String>> getAttestationPolicyWithResponse(AttestationType attestationType) {
+        return withContext(context -> getAttestationPolicyWithResponse(attestationType, context));
+    }
+
+    /**
+     * Retrieves the current policy for an attestation type.
+     *
+     * @param attestationType Specifies the trusted execution environment whose policy should be retrieved.
+     * @param context Context for the remote call.
+     * @throws IllegalArgumentException thrown if parameters fail the validation.
+     * @throws HttpResponseException thrown if the request is rejected by server.
+     * @throws RuntimeException all other wrapped checked exceptions if the request fails to be sent.
+     * @return the attestation policy expressed as a string.
+     */
+    public Mono<Response<String>> getAttestationPolicyWithResponse(AttestationType attestationType, Context context) {
```

Comment:
This can be made package private as it is only used internall by the sync\async client.

Reply:
I thought there needed be a public async `WithResponse` API with a `Context` parameter corresponding to every non-`WithResponse` API. This is the public `WithResponse` API that takes a `Context` parameter.

Is there documentation somewhere which spells out which public variants of an API need to be created? In other words, if I have a `getFoo` API, it spells out which variants of `getFoo` need to be created? I thought it was: `getFoo()`, `getFooWithResponse()`, `getFooWithResponse(Context)`, which is why I have 3 `getAttestationPolicy` API.

Is that not the case?

Reply:
WithResponse(Context)  needs to be made public in non-async client only. 
https://azure.github.io/azure-sdk/java_introduction.html#java-service-client-context
https://azure.github.io/azure-sdk/java_introduction.html#java-service-client-context-async

The way your clients are designed you do need this method (since it is called from AttestationAdministationAsyncClient::getAttestationPolicyWithResponse(AttestationType) and its sync counter part AttestationAdministationClient::getAttestationPolicyWithResponse(AttestationType, Context) but it does not have to be public. We do not expose a public method on Async clients that take Context parameter directly.

Reply:
Ok, that helps a lot. But it brings up a different question: Do I need the `withContext` variants?  Why bother calling `withContext(context -> package_private_method())`? If the `Context` is built into the reactive framework, which circumstances would require generating the context since it's already embedded in the flux/mono.

Reply:
Yes we need them. WithContext under the hood coalesces reactor context to Core context type -  https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/core/azure-core/src/main/java/com/azure/core/util/Context.java

Reply:
Ah - so (as I suspected), the `withContext` method is what stitches the context to the `embedded in the flux/mono` mentioned in the guidelines.